### PR TITLE
[PP-35] Update GitHub actions

### PR DESCRIPTION
**Pull request by Praddby**
Task: https://andrei-lysiuk.atlassian.net/browse/PP-35

Description:
  - add build action for all branches, except main and develop branches
  - add build rule for deploy action.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+          cache: "yarn"
```
Comment (mrakofff) [inline .github/workflows/build.yml]: what is cache? Should it be `node_modules`?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      - run: yarn install
```
Comment (mrakofff) [inline .github/workflows/build.yml]: you could call `yarn` only

**Review by mrakofff — APPROVED** [review]
Some questions but approved

File: .github/workflows/build.yml (hunk 1 above)
Comment (Praddby) [reply .github/workflows/build.yml]: Брал с примера, не тестил, может удалить?) 
А если нам надо будет кэш, уже отдельной таской?)

File: .github/workflows/build.yml (hunk 2 above)
Comment (Praddby) [reply .github/workflows/build.yml]: Ok,

**Pull request merged**